### PR TITLE
Change first/last implementation to prevent redundant comparisons when data is already sorted

### PR DESCRIPTION
**Pull request by mustafasrepo**
## Which issue does this PR close?

Closes #.

## Rationale for this change
This is a follow up PR for [PR8662](https://github.com/apache/arrow-datafusion/pull/8662). With the changes in this PR, first last aggregation does not make comparisons on its data when it is known that their input data already satisfies the ordering requirement.

### CODE REVIEW

**Review by ozankabak — APPROVED** [review]
I reviewed this in detail and it looks good to me. With this, FIRST/LAST functions get the best of both worlds: They make no comparison at all if their inputs are already sorted, but they can work with unsorted data too!

File: datafusion/physical-expr/src/aggregate/first_last.rs [hunk 1]
```diff
@@ -87,6 +90,33 @@ impl FirstValue {
...
         &self.ordering_req
     }
+
+    pub fn with_requirement_satisfied(mut self, requirement_satisfied: bool) -> Self {
+        self.requirement_satisfied = requirement_satisfied;
+        self
+    }
+
+    pub fn convert_to_last(self) -> LastValue {
+        let name = if self.name.starts_with("FIRST") {
```
Comment (comphead) [inline datafusion/physical-expr/src/aggregate/first_last.rs:100]: case sensitive, or it comes upper case always?

Comment (ozankabak) [reply datafusion/physical-expr/src/aggregate/first_last.rs:100]: AFAIK it always comes in uppercase.

**Review by alamb — COMMENTED** [review]
I love it

**Pull request merged**